code/datums/components/thermite.dm

```diff
@@ -0,0 +1,70 @@
+/datum/component/thermite
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	var/static/list/blacklist = typecacheof(/turf/closed/wall/mineral/diamond)
+	var/static/list/resistlist = typecacheof(/turf/closed/wall/r_wall)
+	var/amount
+	var/overlay
+
+/datum/component/thermite/Initialize(_amount)
+	. = ..()
+
+	if(!istype(parent, /turf))
+		qdel(src)
+		return
+	if(blacklist[parent.type])
+		QDEL_IN(src, 0)
+		return
+	if(resistlist[parent.type])
+		_amount*=0.25
+
+	amount = _amount*10
+
+	var/turf/master = parent
+	overlay = mutable_appearance('icons/effects/effects.dmi', "thermite")
+	master.add_overlay(overlay)
+
+	RegisterSignal(COMSIG_COMPONENT_CLEAN_ACT, .proc/clean_react)
+	RegisterSignal(COMSIG_PARENT_ATTACKBY, .proc/attackby_react)
+	RegisterSignal(COMSIG_ATOM_FIRE_ACT, .proc/flame_react)
+
+/datum/component/thermite/Destroy()
+	if(parent)
+		var/turf/master = parent
+		master.cut_overlay(overlay)
+	return ..()
+
+/datum/component/thermite/InheritComponent(datum/component/thermite/newC, i_am_original)
+	if(!i_am_original)
+		return
+	var/_amount = newC.amount
+	if(resistlist[parent.type])
+		_amount*=0.25
+	amount += _amount*10
+
+/datum/component/thermite/proc/thermite_melt(mob/user)
+	var/turf/master = parent
+	master.cut_overlay(overlay)
+	var/obj/effect/overlay/thermite/fakefire = new(master)
+
+	playsound(master, 'sound/items/welder.ogg', 100, 1)
+
+	if(amount >= 50)
+		var/burning_time = max(100, 100-amount)
+		master = master.ChangeTurf(master.baseturf)
+		master.burn_tile()
+		if(user)
+			master.add_hiddenprint(user)
+		QDEL_IN(fakefire, burning_time)
+	else
+		QDEL_IN(fakefire, 50)
+
+/datum/component/thermite/proc/clean_react(strength)
+	qdel(src)
```

Comment:
Regardless of strength?

Reply:
That's how it was before and I was mostly keeping thermite the same functionality wise